rpc/client.go

```diff
@@ -60,7 +60,7 @@ const (
 	// The approach taken here is to maintain a per-subscription linked list buffer
 	// shrinks on demand. If the buffer reaches the size below, the subscription is
 	// dropped.
-	maxClientSubscriptionBuffer = 8000
+	maxClientSubscriptionBuffer = 20000
```

Comment:
I am not sure we want to increase this. If we don't increase it however, we have to run our tests with low number of messages.

Reply:
can we make it a parameter?

Reply:
I've just pushed a commit for that. To be honest, I am not sure parameterising this is worth it, I think we are just increasing complexity. We have to think if this makes sense in a real-world scenario. What do you think?

Reply:
Parameterising this gets confusing fast, because different clients are initialised at different places in the codebase, and would have different buffer sizes.

I think it would be better if we do one of the following:
1. Make the const global (or have a global modifier), so that we can change it globally for the whole codebase for the purpose of testing.
2. Use a "reasonable" value that works for the tests we want to run.
3. Update the functionality of rpc/subscription and make this buffer resizable if possible.

I think all those 3 solutions are better than the one I had added, where each client can modify its subscription queue, therefore I reverted the commit.

Reply:
3. sounds like the better bet to me?

Reply:
Actually looking at rpc/subscription - we are using a linked list, which does not have capacity. This `max limit` appears to be just a sanity check / upper-bound, so to me it makes sense going with 1 or 2.

Reply:
Ok. I vote for 1) in this case.

Reply:
I think it would be best to discuss this with @fjl on how this parameter was chosen in the first place. Probably best not to introduce too many changes to the core go-ethereum packages, considering how many outstanding PRs and code we have for Swarm. Can we keep this value as is for now, and address it's change when we are closer to merge time with `pss-networktest`, `swarm-network-rewrite` and `pss`?

Reply:
Out of curiosity; did you discuss this with @fjl in the end? Or are we still deferring the final nature of the fix?

Reply:
No, I still haven't done that.

It is a well understood change, and I will just confirm the reason for choosing that value once, we get to the point of merging this to `go-ethereum/master`.

You could argue that what we're doing is a stress test and if we want to keep tests fast and light, maybe we should not run tests that hit this threshold as part of the default Travis test suite.

Reply:
Yeah either that or maybe have the option of running individual parametric tests from the command line for the heavy hitting.

I recently installed `monero` - they have a separate test in the `Makefile` that takes half an hour to run.